src/hooks/useProfileModal.ts

```diff
@@ -0,0 +1,27 @@
+import { useCallback } from "react";
+
+import { userProfileSelector } from "utils/selectors";
+
+import { setUserProfileData } from "store/actions/UserProfile";
+
+import { useSelector } from "./useSelector";
+import { useDispatch } from "./useDispatch";
+import { User } from "types/User";
+import { WithId } from "utils/id";
+
+export const useProfileModal = () => {
+  const dispatch = useDispatch();
+  const selectedUserProfile = useSelector(userProfileSelector);
+
+  const setUserProfile = useCallback(
+    (userProfile: WithId<User> | undefined) => {
```

Comment:
We use the `?` shorthand rather than explicit `| undefined` in params like this:
```suggestion
  const setUserProfile = useCallback(
    (userProfile?: WithId<User>) => {
```

Reply:
It might be worth including a `useCallback`'d `unsetUserProfile` here as well.

Reply:
Which it seems @mike-lvov already mentioned in https://github.com/sparkletown/sparkle/pull/1218/files#r584710069